pkg/kubelet/cm/memorymanager/memory_manager.go

```diff
@@ -176,38 +177,31 @@ func (m *manager) Start(activePods ActivePodsFunc, sourcesReady config.SourcesRe
 }
 
 // AddContainer saves the value of requested memory for the guaranteed pod under the state and set memory affinity according to the topolgy manager
-func (m *manager) AddContainer(pod *v1.Pod, container *v1.Container, containerID string) error {
+func (m *manager) AddContainer(pod *v1.Pod, container *v1.Container, containerID string) {
 	m.Lock()
+	defer m.Unlock()
+
 	m.containerMap.Add(string(pod.UID), container.Name, containerID)
-	m.Unlock()
+}
 
+// GetMemory provides NUMA nodes that used to allocate the container memory
+func (m *manager) GetMemoryNUMANodes(pod *v1.Pod, container *v1.Container) sets.Int {
 	// Get NUMA node affinity of blocks assigned to the container during Allocate()
-	var nodes []string
+	numaNodes := sets.NewInt()
 	for _, block := range m.state.GetMemoryBlocks(string(pod.UID), container.Name) {
 		for _, nodeID := range block.NUMAAffinity {
-			nodes = append(nodes, strconv.Itoa(nodeID))
+			// avoid nodes duplication when hugepages and memory blocks pinned to the same NUMA node
+			numaNodes.Insert(nodeID)
 		}
 	}
 
-	if len(nodes) < 1 {
-		klog.V(5).Infof("[memorymanager] update container resources is skipped due to memory blocks are empty")
-		return nil
+	if len(numaNodes) < 1 {
```

Comment:
I would just do `== 0` instead of `< 1` here.
I would also explicitly return nil, to make it obvious that this is the exceptional case with no elements.

Reply:
done